src/Microsoft.ComponentDetection.Detectors/pip/PipReportComponentDetector.cs

```diff
@@ -200,7 +229,7 @@ protected override async Task OnFileFoundAsync(ProcessRequest processRequest, ID
                 // futures; python_version <= \"2.7\"
                 // sphinx (!=1.8.0,!=3.1.0,!=3.1.1,>=1.6.5) ; extra == 'docs'
                 var dependencySpec = new PipDependencySpecification($"Requires-Dist: {dependency}", requiresDist: true);
-                if (dependencySpec.PackageIsUnsafe())
+                if (!dependencySpec.IsValidParentPackage(this.pythonResolver.GetPythonEnvironmentVariables()))
```

Comment:
Can we avoid calling `this.pythonResolver.GetPythonEnvironmentVariables()` every iteration?